src/libs/common/ide/types/ide.types.ts

```diff
@@ -24,17 +26,49 @@ export interface IDE {
    * The root directory of this shipped code.  Can be used to access bundled
    * assets.
    */
-  assetsRoot: string;
+  readonly assetsRoot: string;
 
   /**
    * Whether we are running in development, test, or production
    */
-  runMode: RunMode;
+  readonly runMode: RunMode;
 
   /**
    * A list of workspace folders for the currently active workspace
    */
-  workspaceFolders: readonly WorkspaceFolder[] | undefined;
+  readonly workspaceFolders: readonly WorkspaceFolder[] | undefined;
+
+  /**
+   * The currently active editor or `undefined`. The active editor is the one
+   * that currently has focus or, when none has focus, the one that has changed
+   * input most recently.
+   */
+  readonly activeTextEditor: TextEditor | undefined;
+
+  /**
+   * Same as {@link activeTextEditor} but editable
+   */
+  readonly activeEditableTextEditor: EditableTextEditor | undefined;
+
+  /**
+   * The currently visible editors or an empty array.
+   */
+  readonly visibleTextEditors: TextEditor[];
+
+  /**
+   * Get an editable version of the text editor.
+   * @param editor A editable text editor
+   */
+  getEditableTextEditor(editor: TextEditor): EditableTextEditor;
+
+  /**
+   * An event that is emitted when a {@link TextDocument text document} is changed. This usually happens
+   * when the {@link TextDocument.getText contents} changes but also when other things like the
+   * {@link TextDocument.isDirty dirty}-state changes.
+   */
+  onDidChangeTextDocument(
+    listener: (event: TextDocumentChangeEvent) => void,
+  ): Disposable;
```

Comment:
Tbh I'm a bit skeptical about mirroring their whole `onDidXxx` returning a `Disposable` setup, but I think ok to parrot at least some of their interface for the first draft of our api spec

Reply:
Agree that this could probably be cleaned up. Just for now I try to get this pr to not balloon anymore.